Lib/test/support/script_helper.py

```diff
@@ -84,6 +85,21 @@ def fail(self, cmd_line):
                                 err))
 
 
+def run_until(condition, timeout=60, error_msg=None):
+    """Evaluates a condition until it becomes True, or a timeout is reached."""
+    deadline = time.monotonic() + timeout
+    t = 0.1
+    while time.monotonic() < deadline:
```

Comment:
For infinitesimal `timeout` values, the `while` loop will never be entered, and `condition` will not be evaluated even one single time. Not such a big deal, but maybe worth noting.